diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index f29ee7a..64bd5b7 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -424,9 +424,11 @@ else()
         mark_as_advanced(APPLE_STATIC_LOADER)
     endif()
 
-    if(APPLE_STATIC_LOADER)
+    if(APPLE_STATIC_LOADER OR BUILD_STATIC_LOADER)
         add_library(vulkan STATIC)
-        target_compile_definitions(vulkan PRIVATE APPLE_STATIC_LOADER)
+        if(APPLE_STATIC_LOADER)
+            target_compile_definitions(vulkan PRIVATE APPLE_STATIC_LOADER)
+        endif()
 
         message(WARNING "The APPLE_STATIC_LOADER option has been set. Note that this will only work on MacOS and is not supported "
                 "or tested as part of the loader. Use it at your own risk.")
@@ -544,9 +546,9 @@ if (BUILD_TESTS)
     target_compile_definitions(vulkan PRIVATE SHOULD_EXPORT_TEST_FUNCTIONS)
 endif()
 
-if (APPLE_STATIC_LOADER)
+if (APPLE_STATIC_LOADER OR BUILD_STATIC_LOADER)
     # TLDR: This feature only exists at the request of Google for Chromium. No other project should use this!
-    message(NOTICE "Apple STATIC lib: it will be built but not installed, and vulkan.pc and VulkanLoaderConfig.cmake won't be generated!")
+    message(NOTICE "STATIC lib: it will be built but not installed, and vulkan.pc and VulkanLoaderConfig.cmake won't be generated!")
     return()
 endif()
 
